File: front/apps/web-ele/src/api/core/production_order.ts
import { requestClient } from '../request';

export interface ProductionOrder {
  id: number;
  productionCode: string;
  orderCode: string;
  processCode: string;
  supplierCode: string;
  productionQty: number;
  productionStatus: string;
  createdBy: number;
  createdAt: string;
  updatedAt: string;
}

export interface ProductionOrderListResponse {
  total: number;
  list: ProductionOrder[];
}

export async function createProductionOrder(data: {
  productionCode: string;
  orderCode: string;
  processCode: string;
  supplierCode: string;
  productionQty: number;
  productionStatus: string;
  createdBy: number;
}) {
  const response = await requestClient.post<ProductionOrder>('/production_orders', data);
  return response;
}

export async function getProductionOrders(params?: { limit?: number; offset?: number }) {
  const response = await requestClient.get<ProductionOrderListResponse>('/production_orders', {
    params: {
      limit: params?.limit || 100,
      offset: params?.offset || 0,
    },
  });
  return response;
}

export async function getProductionOrder(id: number) {
  const response = await requestClient.get<ProductionOrder>(`/production_orders/${id}`);
  return response;
}

export async function updateProductionOrder(
  id: number,
  data: {
    productionCode?: string;
    orderCode?: string;
    processCode?: string;
    supplierCode?: string;
    productionQty?: number;
    productionStatus?: string;
  }
) {
  const response = await requestClient.post<ProductionOrder>(`/production_orders/${id}`, data);
  return response;
}

export async function deleteProductionOrder(id: number) {
  const response = await requestClient.delete(`/production_orders/${id}`);
  return response;
}

File: back/internal/production_order/domain/production_order.go
package domain

import (
	"time"

	"gorm.io/gorm"
)

// ProductionOrder 生产指示
type ProductionOrder struct {
	ID        uint           `gorm:"primaryKey" json:"id"`
	ProductionCode string `gorm:"size:50;uniqueIndex;not null" json:"productionCode"`
	OrderCode string `gorm:"size:50;not null;index" json:"orderCode"`
	ProcessCode string `gorm:"size:50;not null;index" json:"processCode"`
	SupplierCode string `gorm:"size:50;not null;index" json:"supplierCode"`
	ProductionQty float64 `gorm:"type:decimal(10,2);not null" json:"productionQty"`
	ProductionStatus string `gorm:"size:50;not null;default:'pending'" json:"productionStatus"`
	CreatedBy string         `gorm:"size:50;not null" json:"createdBy"`
	CreatedAt time.Time      `gorm:"autoCreateTime" json:"createdAt"`
	UpdatedAt time.Time      `gorm:"autoUpdateTime" json:"updatedAt"`
	DeletedAt gorm.DeletedAt `gorm:"index" json:"deletedAt,omitempty"`
}

// TableName 表名
func (ProductionOrder) TableName() string {
	return "production_orders"
}
